src/interfaces/ecpg/test/pgtypeslib/meson.build

```diff
@@ -0,0 +1,29 @@
+pgc_files = [
+  'dt_test',
+  'dt_test2',
+  'num_test',
+  'num_test2',
+  'nan_test',
+]
+
+foreach pgc_file: pgc_files
+  exe_input = custom_target(pgc_file,
+    input: '@0@.pgc'.format(pgc_file),
+    output: '@0@.c'.format(pgc_file),
+    command: [
+      ecpg_exe, 
+      '--regression',
+      '-I' + meson.current_source_dir(),
```

Comment:
The source-dir reference should probably replaced by `@CURRENT_SOURCE_DIR@` and `@SOURCE_ROOT@`.

The output file `.format()` could be replaced with `@PLAINNAME@`.

With that the entire command shouldn't differ between the directories. That should allow nearly all the arguments to be specified as a single kwargs: ecpg_preproc_test_kwargs?

Reply:
These custom targets have different flags in commands like `'-C', 'INFORMIX'`, `'-C', 'ORACLE'`, `'-r', 'prepare'`, `'-i'`, instead I can create variable(ecpg_preproc_test_command) for same parts of the commands.